src/notifications/settings/initialization.ts

```diff
@@ -0,0 +1,192 @@
+import {
+  DEFAULT_ENABLED_TOPIC_SETTINGS,
+  NotificationRelationship,
+  NOTIFICATIONS_DEFAULT_CHAIN_ID,
+  WALLET_GROUPS_STORAGE_KEY,
+} from '@/notifications/settings/constants';
+import {
+  AddressWithRelationship,
+  WalletNotificationSettings,
+} from '@/notifications/settings/types';
+import {
+  getAllNotificationSettingsFromStorage,
+  notificationSettingsStorage,
+  setAllNotificationSettingsToStorage,
+} from '@/notifications/settings/storage';
+import { notificationsSubscription } from '@/redux/explorer';
+import { AppDispatch } from '@/redux/store';
+import {
+  subscribeWalletToAllEnabledTopics,
+  unsubscribeWalletFromAllNotificationTopics,
+} from '@/notifications/settings/firebase';
+import { InteractionManager } from 'react-native';
+import { logger, RainbowError } from '@/logger';
+import { removeNotificationSettingsForWallet } from '@/notifications/settings/settings';
+
+/**
+ Checks if group notification settings are present in storage
+ and adds default values for them if they do not exist.
+ */
+export const addDefaultNotificationGroupSettings = () => {
+  const data = notificationSettingsStorage.getString(WALLET_GROUPS_STORAGE_KEY);
+
+  if (!data) {
+    const defaultSettings = {
+      [NotificationRelationship.OWNER]: true,
+      [NotificationRelationship.WATCHER]: false,
+    };
+    notificationSettingsStorage.set(
+      WALLET_GROUPS_STORAGE_KEY,
+      JSON.stringify(defaultSettings)
+    );
+  }
+};
+
+/**
+ * Adds fresh disabled settings for all wallets that didn't have settings
+ */
+export const initializeNotificationSettingsForAddresses = (
```

Comment:
Scared of this whole file tbh. Could we write some unit tests for it?

Reply:
Do you have ideas how to make it better so that you won't be scared?